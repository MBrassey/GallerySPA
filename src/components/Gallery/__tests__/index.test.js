// __tests__/Gallery.test.js
import React from 'react'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Gallery from '..'
const architecture = { name: "architecture", description: "Accent of architecture" };


afterEach(cleanup)

describe('Gallery component', () => {

  it('renders', () => {
    render(<Gallery currentCategory={architecture} />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Gallery currentCategory={architecture} />)
    expect(asFragment()).toMatchSnapshot()
  })
});

it('displays text', () => {
  const { getByTestId } = render(<Gallery currentCategory={architecture} />)
  expect(getByTestId('h1tag')).toHaveTextContent('Architecture')
})