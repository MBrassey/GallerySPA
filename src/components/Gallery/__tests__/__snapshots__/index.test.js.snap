// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="h1tag"
    >
      Architecture
    </h1>
    <p>
      Accent of architecture
    </p>
    <div>
      <div
        class="flex-row"
      >
        <img
          alt="TriTunnel"
          class="img-thumbnail mx-1"
          src="0.jpg"
        />
        <img
          alt="Balconies"
          class="img-thumbnail mx-1"
          src="1.jpg"
        />
        <img
          alt="Hex"
          class="img-thumbnail mx-1"
          src="2.jpg"
        />
        <img
          alt="Modern Home"
          class="img-thumbnail mx-1"
          src="3.jpg"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
